x-pack/legacy/plugins/fleet/public/pages/agent_list/index.tsx

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState, useEffect } from 'react';
+import {
+  EuiInMemoryTable,
+  EuiPageBody,
+  EuiPageContent,
+  EuiTitle,
+  EuiSpacer,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiEmptyPrompt,
+  EuiLink,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { FrontendLibs } from '../../lib/types';
+import { AgentHealth } from '../../components/agent_health';
+
+interface RouterProps {
+  libs: FrontendLibs;
+}
+
+export const AgentListPage: React.SFC<RouterProps> = ({ libs }) => {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [agents, setAgents] = useState<any[]>([]);
+
+  const fetchAgents = async () => {
+    setIsLoading(true);
+    setAgents(await libs.agents.getAll());
+    setIsLoading(false);
+  };
+
+  // Load agents
+  useEffect(() => {
```

Comment:
We should prob use a timer, and poll every X seconds

Reply:
agreed, I will change this when I do the refactoring work to support manual pagination. with EuiInMemoryTable, refreshing data resets the table state!

Reply:
ah true I forgot it did that